src/kili/queries/asset/helpers.py

```diff
@@ -49,133 +49,130 @@ def get_file_extension_from_headers(url) -> Optional[str]:
-        local_path = local_path.with_suffix(extension)
-    return local_path.resolve()
+    filename = external_id
+    if extension is not None and not filename.endswith(extension):
+        filename = filename + extension
+    local_dir_path = local_dir_path / filename
+    return local_dir_path.resolve()
 
 
 @retry(stop=stop_after_attempt(2), wait=wait_random(min=1, max=2))
-def download_file(url: str, external_id: str, is_video_frame: bool, local_dir_path: Path) -> Path:
+def download_file(url: str, external_id: str, local_dir_path: Path) -> str:
     """
     Download a file by streming chunks of 1Mb
     If the file already exists in local, it does not download it
     """
-    local_path = get_download_path(url, external_id, is_video_frame, local_dir_path)
+    local_path = get_download_path(url, external_id, local_dir_path)
     local_path.parent.mkdir(parents=True, exist_ok=True)
     if not local_path.is_file():
         with requests.get(url, stream=True, timeout=20) as response:
             response.raise_for_status()
             with open(local_path, "wb") as file:
                 for chunk in response.iter_content(chunk_size=1024 * 1024):
                     file.write(chunk)
-    return local_path
+    return str(local_path)
 
 
-def assert_required_fields_existence(asset: dict) -> None:
-    """Get a list of an asset field from a generator of kili assets"""
-    required_fields = ["content", "externalId", "jsonContent", "project.inputType"]
+def assert_required_fields_existence(assets: List[Dict]) -> None:
+    """check if all fields are available to download assets"""
+    required_fields = ["content", "externalId"]
     for field in required_fields:
-        if field not in asset.keys():
+        if field not in assets[0].keys():
             raise MissingPropertyError(
                 f"The asset does not have the {field} field. Please add it to the fields when"
                 " querying assets."
             )
 
+    if "jsonContent" not in assets[0].keys():
+        warnings.warn(
+            "It is recommended to add 'jsonContent' and 'project.inputType' to the 'fields'"
+            " argument in order to download all types of content."
+        )
+
+    else:
+        has_non_empty_jsoncontent = any(asset["jsonContent"] != "" for asset in assets)
+        if has_non_empty_jsoncontent:
+            try:
+                assets[0]["project"]["inputType"]
+            except KeyError as exc:
+                raise MissingPropertyError(
+                    "To download 'jsonContent' assets, please add 'project.inputType' to the fields"
+                    " when querying assets."
+                ) from exc
+
+
+def download_single_asset(asset: Dict, local_dir_path: Path) -> Dict[str, Any]:
+    """Download single asset on disk and modify asset attributes"""
+
+    if asset["content"] != "":
+        asset["content"] = download_file(asset["content"], asset["externalId"], local_dir_path)
```

Comment:
In fact here it will download both, the frames and the original video. Don't know if we should do this

Reply:
you're right yes it will download both.

Maybe we will need the frames for the yolo, voc, exports no?

Reply:
If we only download one among the two, it will be the frames and not the content. because the user annotate on the frames that we have split from the video.

Reply:
I do not see a use case where the content would be more useful than the frames, so I am for outputting the frames.